### PR TITLE
Feature/max txns per round

### PR DESCRIPTION
**Pull request by talekhinezh**
* Implement ledger level transaction limits on proposals

### CODE REVIEW

**Review by michalzajda — APPROVED** [review]

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/SystemConfigService.java [hunk 1]
```diff
@@ -350,22 +348,21 @@ static JSONObject prepareApiConfiguration(List<EndpointStatus> statuses) {
...
-		int maxTxnsPerProposal
+		int maxValidators
 	) {
 		var forks = jsonArray();
 		forkConfigTreeMap.forEach((e, config) -> forks.put(
 			jsonObject()
 				.put("name", config.getName())
 				.put("version", config.getVersion().name().toLowerCase())
 				.put("maxRounds", config.getConfig().getMaxRounds())
+				.put("maxSigsPerRound", config.getConfig().getMaxSigsPerRound().orElse(-1))
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/SystemConfigService.java:359]: This change needs to be propagated to open rpc spec

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/SystemConfigService.java [hunk 2]
```diff
@@ -350,22 +348,21 @@ static JSONObject prepareApiConfiguration(List<EndpointStatus> statuses) {
...
 				.put("epoch", e)
 		));
 
 		return jsonObject()
 			.put("minValidators", minValidators)
 			.put("maxValidators", maxValidators)
-			.put("maxTxnsPerProposal", maxTxnsPerProposal)
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/service/SystemConfigService.java:368]: This change needs to be propagated to open rpc spec

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineModule.java [hunk 3]
```diff
@@ -71,6 +72,14 @@ private REParser parser(RERules rules) {
 		return rules.getParser();
 	}
 
+	// TODO: Remove
+	@Provides
+	@Singleton
+	@MaxTxnsPerProposal
+	private OptionalInt maxTxnsPerProposal(RERules rules) {
+		return rules.getMaxTxnsPerRound();
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineModule.java:80]: Is it still used?

**Review by siy — APPROVED** [review]
LGTM with minor comments

**Pull request merged**